test/internals/initialConfig.test.js

```diff
@@ -367,3 +367,54 @@ test('pluginTimeout should be parsed correctly', t => {
   t.equal(withInvalidTimeout.initialConfig.pluginTimeout, 10000)
   t.end()
 })
+
+test('Should not have issues when using same options object for multiple Fastify instance', async t => {
```

Comment:
I don't understand how this test proves the PR does what it is supposed to do.

Reply:
I use 2 instances in this test in order to check if there is an error.
Thanks to @seriousme from #5702.

Reply:
Why not just test that the object passed is not mutated?

Reply:
> Why not just test that the object passed is not mutated?

Yes, It was so obvious 🥲, thanks.
I will clean this test.